.github/workflows/version.yml

```diff
@@ -0,0 +1,50 @@
+name: Increment Version
+
+on:
+  push:
+    tags:
+      - '*.*.*'
+
+permissions: {}
+jobs:
+  build:
+    if: github.repository == 'opensearch-project/opensearch-rs'
+    runs-on: ubuntu-latest
+    steps:
+      - name: GitHub App token
+        id: github_app_token
+        uses: tibdex/github-app-token@v2.1.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+          installation_id: 22958780
+
+      - uses: actions/checkout@v4
+      - name: Fetch and Update Version Information
+        run: |
+          TAG=$(echo "${GITHUB_REF#refs/*/}")
+          CURRENT_VERSION_ARRAY=($(echo "$TAG" | tr . '\n'))
+          CURRENT_VERSION=$(IFS=. ; echo "${CURRENT_VERSION_ARRAY[*]:0:3}")
+          CURRENT_VERSION_ARRAY[2]=$((CURRENT_VERSION_ARRAY[2]+1))
+          NEXT_VERSION=$(IFS=. ; echo "${CURRENT_VERSION_ARRAY[*]:0:3}")
+          echo "TAG=$TAG" >> $GITHUB_ENV
+          echo "CURRENT_VERSION=$CURRENT_VERSION" >> $GITHUB_ENV
+          echo "NEXT_VERSION=$NEXT_VERSION" >> $GITHUB_ENV
+          sed -i '' -e "s/^version = \"$CURRENT_VERSION\"/version = \"$NEXT_VERSION\"/g" opensearch/Cargo.toml
```

Comment:
This assumes that `$CURRENT_VERSION` is the recently released version and is what is already in Cargo.toml

Reply:
or will it make sense to do a wildcard match:
```
sed -i '' -e "s/^version = \".*\"/version = \"$NEXT_VERSION\"/g" opensearch/Cargo.toml
```